database/misc/redis.py

```diff
@@ -257,15 +253,12 @@ def main():
 
     # flush Command section -----------
     elif command == "flush":
-        try:
-            db = int(module.params['db'])
-        except Exception:
-            db = 0
+        db = module.params['db']
         mode = module.params['flush_mode']
 
         #Check if we have all the data
         if mode == "db":
-            if type(db) != int:
+            if db is None:
                 module.fail_json(
```

Comment:
These last two sections of the original code confuse me a little.  It looks like the first section is saying "Make the db parameter into an integer.  If, for any reason, we can't turn it into an integer then set db to 0.  Then the second section is unnecessarily making a check whether db  really is an int now.

Probably those can just be translated into a default value of 0 in the argument spec and then we don't have to check whether db is None here.